hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/StoreFileWriter.java

```diff
@@ -256,156 +234,571 @@ public void appendMobMetadata(SetMultimap<TableName, String> mobRefSet) throws I
   public void appendTrackedTimestampsToMetadata() throws IOException {
     // TODO: The StoreFileReader always converts the byte[] to TimeRange
     // via TimeRangeTracker, so we should write the serialization data of TimeRange directly.
-    appendFileInfo(TIMERANGE_KEY, TimeRangeTracker.toByteArray(timeRangeTracker));
-    appendFileInfo(EARLIEST_PUT_TS, Bytes.toBytes(earliestPutTs));
+    liveFileWriter.appendTrackedTimestampsToMetadata();
```

Comment:
OK, I think the metadata is a missing part in our previous designs...

In general, we should support writing multiple files for the DefaultStoreFileWriter, but then, we need to track some metadatas separately, for the different files, like timestamp range, cells count, etc.

Reply:
I did not change the existing behavior. I refactored the code such that the exiting StoreFileWriter implementation was moved to the new nested class called SingleStoreFileWriter. StoreFileWriter has now two SingleStoreFileWriter objects, liveFileWriter, and historicalFileWriter.

Reply:
Checked the code, besides calling the (old) StoreFileWriter, this method is only called in HFileOutputFormat2, for appending metadata for store files generated by MR job. Seems better to introduce a special method in StoreFileWriter for it. Anyway, not a problem for this PR. Can file another issue for it.